notifications/templates.ts

```diff
@@ -0,0 +1,45 @@
+import { MessageType } from './messageTypes';
+import { GraphQLUser } from '../common/user/session';
+import { Concrete_notification as ConcreteNotification } from '../graphql/generated';
+
+// only actionPath or url is optionally allowed but not both
+type NotificationAction = { navigateTo?: string; url?: undefined } | { url: string; navigateTo?: undefined };
+export type MessageTemplate = {
+    header: string;
+    body: string;
+    messageType: MessageType;
+} & NotificationAction;
+
+export const getMessage = (concreteNotification: ConcreteNotification, user: GraphQLUser): MessageTemplate => {
+    const { firstname, lastname } = user;
+    const { notificationID, context } = concreteNotification;
+
+    const templates: { [key: number]: MessageTemplate } = {
+        1: {
+            header: `bla bla ${firstname}`,
+            body: `bla bla ${firstname} bla bla ${lastname}`,
+            navigateTo: `somepath`,
+            messageType: MessageType.APPOINTMENT,
+        },
+        2: {
+            header: `bla bla ${firstname}`,
+            body: `bla bla ${firstname} bla bla ${lastname}`,
+            navigateTo: 'welcome',
+            messageType: MessageType.APPOINTMENT,
+        },
+        3: {
+            header: `bla bla ${firstname}`,
+            body: `bla bla ${firstname} bla bla ${lastname}`,
+            messageType: MessageType.MATCH,
+        },
+        // 0 used as fallback
+        0: {
+            header: `Message not found`,
+            body: `Error: message details not found. Message for ${firstname} ${lastname}`,
+            messageType: MessageType.MATCH,
+            navigateTo: 'welcome',
+        },
+    };
+    const index = templates.hasOwnProperty(notificationID) ? notificationID : 0;
```

Comment:
Uh, instead of a magic number "0" I'd prefer a separate constant for the error message, then `templates[notificationID] ?? MESSAGE_NOT_FOUND`.

Reply:
the default message was more for testing purposes, because I was not sure which notifications I would get

Reply:
changed error handling